### PR TITLE
NAS-132060 / 25.04 / Remove get/set immutable

### PR DESCRIPTION
**Pull request by anodos325**
filesystem.get_immutable and filesystem.set_immutable are redundant endpoints since we now return statx attrs in filesystem.stat and filesystem.listdir output and can set generic ZFS attributes via the dedicated endpoint.

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]

File: tests/api2/test_300_nfs.py [hunk 1]
```diff
@@ -222,9 +222,13 @@ def set_immutable_state(path: str, want_immutable=True):
     '''
     Used by exportsd test
     '''
-    call('filesystem.set_immutable', want_immutable, path)
-    res = call('filesystem.is_immutable', '/etc/exports.d')
-    assert res is want_immutable, f"Expected mutable filesystem: {res}"
+    call('filesystem.set_zfs_attributes', {
+        'path': d,
+        'zfs_file_attributes': {'immutable': flag_set}
```
Comment (mgrimesix) [inline tests/api2/test_300_nfs.py:227]: `flag_set`?   Maybe `want_immutable`?

File: src/middlewared/middlewared/plugins/account.py [hunk 2]
```diff
@@ -2027,5 +2028,13 @@ async def __common_validation(self, verrors, data, schema, pk=None):
 
 
 async def setup(middleware):
+    try:
+        await middleware.call('filesystem.set_zfs_attributes', {
+            'path': DEFAULT_HOME_PATH,
+            'zfs_file_attributes': {'immutable': True}
+        })
+    except Exception:
+        middleware.logger.error('Failed to set immmutable property on /var/empty', exc_info=True)
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/account.py:2037]: First typo: immmutable -> immutable

Second, I know it equates to the same thing, but just to future-proof, would this be better?
```
 middleware.logger.error('Failed to set immutable property on %r', DEFAULT_HOME_PATH, exc_info=True)

```

File: src/middlewared/middlewared/plugins/smb.py [hunk 3]
```diff
@@ -368,18 +368,6 @@ async def configure(self, job, create_paths=True):
         job.set_progress(20, 'Wait for ix-netif completion.')
         await self.netif_wait()
 
-        """
-        Since some NSS modules will default to setting home directory to /var/empty,
-        verify that this path is immutable during setup for SMB service (prior to
-        initializing directory services).
-        """
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/smb.py:375]: Looks like this is now being done in account.py `setup`.  Maybe move this comment there (and adapt)?

File: tests/api2/test_300_nfs.py [hunk 4]
```diff
@@ -222,9 +222,13 @@ def set_immutable_state(path: str, want_immutable=True):
...
-    call('filesystem.set_immutable', want_immutable, path)
-    res = call('filesystem.is_immutable', '/etc/exports.d')
-    assert res is want_immutable, f"Expected mutable filesystem: {res}"
+    call('filesystem.set_zfs_attributes', {
+        'path': path,
+        'zfs_file_attributes': {'immutable': want_immutable}
+    })
+    is_immutable = 'IMMUTABLE' in call('filesystem.stat', '/etc/exports.d')['attributes']
+
+    assert is_immutable is want_immutable, f"Expected mutable filesystem: {is_immutable}"
```
Comment (bmeagherix) [inline tests/api2/test_300_nfs.py:231]: Text of message should really depend on value of `want_immutable`.  (Meh.)

Comment (mgrimesix) [reply tests/api2/test_300_nfs.py:231]: Correct!  That should be `want_immutable`

Comment (bmeagherix) [reply tests/api2/test_300_nfs.py:231]: Or
`assert is_immutable is want_immutable, f"Expected {'im' if want_immutable else ''}mutable filesystem: {is_immutable}"`

:-)

Comment (mgrimesix) [reply tests/api2/test_300_nfs.py:231]: Not necessary actually.   The failing assert will be output and the helpful 'expected' message.  What else is needed to see what failed on the test?

**Review by mgrimesix — APPROVED** [review]
LGTM!

**Review by bmeagherix — APPROVED** [review]

**Pull request merged**